engine/src/main/java/com/google/android/fhir/search/MoreSearch.kt

```diff
@@ -169,6 +183,39 @@ val Order?.sqlString: String
       null -> ""
     }
 
+private fun getConditionParamPair(
+  prefix: ParamPrefixEnum,
+  value: DateTimeType
+): Pair<String, List<Any>> {
+  return when (prefix) {
+    ParamPrefixEnum.APPROXIMATE -> TODO("Not Implemented")
+    ParamPrefixEnum.STARTS_AFTER ->
+      "? <= index_from" to listOf(value.precision.add(value.value, 1).time)
+    ParamPrefixEnum.ENDS_BEFORE -> "? >= index_to" to listOf(value.value.time)
+    ParamPrefixEnum.NOT_EQUAL ->
+      "index_from NOT BETWEEN ? AND ? OR index_to NOT BETWEEN ? AND ?" to
+        listOf(
+          value.value.time,
+          value.precision.add(value.value, 1).time - 1,
+          value.value.time,
+          value.precision.add(value.value, 1).time - 1
+        )
+    ParamPrefixEnum.EQUAL ->
+      "index_from BETWEEN ? AND ? AND index_to BETWEEN ? AND ?" to
+        listOf(
+          value.value.time,
+          value.precision.add(value.value, 1).time - 1,
+          value.value.time,
+          value.precision.add(value.value, 1).time - 1
+        )
+    ParamPrefixEnum.GREATERTHAN ->
+      "? <= index_from" to listOf(value.precision.add(value.value, 1).time)
```

Comment:
```suggestion
      "? <= index_to" to listOf(value.precision.add(value.value, 1).time)
```

see this: `[parameter]=gt2013-01-14T10:00 | 2013-01-14 matches, because it includes the part of 14-Jan 2013 after 10am`

what this says to me is that if any part of the range is greater than the given value, the range is considered greater than the given value.

same for ge